soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +216,115 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
+    struct CowInfo cowInfo;
+
+    switch (globalCtx->sceneNum) {
+        case SCENE_SOUKO: // Lon Lon Tower
+            if (this->actor.world.pos.x == -229 && this->actor.world.pos.z == 157) {
+                cowInfo.cowId = 0;
+                cowInfo.randomizerCheck = RC_LLR_TOWER_LEFT_COW;
+            } else {
+                cowInfo.cowId = 1;
+                cowInfo.randomizerCheck = RC_LLR_TOWER_RIGHT_COW;
+            }
+            break;
+        case SCENE_MALON_STABLE:
+            if (this->actor.world.pos.x == 116 && this->actor.world.pos.z == -254) {
+                cowInfo.cowId = 2;
+                cowInfo.randomizerCheck = RC_LLR_STABLES_RIGHT_COW;
+            } else {
+                cowInfo.cowId = 3;
+                cowInfo.randomizerCheck = RC_LLR_STABLES_LEFT_COW;
+            }
+            break;
+        case SCENE_KAKUSIANA: // Grotto
+            if (this->actor.world.pos.x == 2444 && this->actor.world.pos.z == -471) {
+                cowInfo.cowId = 4;
+                cowInfo.randomizerCheck = RC_DMT_COW_GROTTO_COW;
+            } else {
+                cowInfo.cowId = 5;
+                cowInfo.randomizerCheck = RC_HF_COW_GROTTO_COW;
+            }
+            break;
+        case SCENE_LINK_HOME:
+            cowInfo.cowId = 6;
+            cowInfo.randomizerCheck = RC_KF_LINKS_HOUSE_COW;
+            break;
+        case SCENE_LABO: // Impa's house
+            cowInfo.cowId = 7;
+            cowInfo.randomizerCheck = RC_KAK_IMPAS_HOUSE_COW;
+            break;
+        case SCENE_SPOT09: // Gerudo Valley
+            cowInfo.cowId = 8;
+            cowInfo.randomizerCheck = RC_GV_COW;
+            break;
+        case SCENE_SPOT08: // Jabu's Belly
+            cowInfo.cowId = 9;
+            cowInfo.randomizerCheck = RC_JABU_JABUS_BELLY_MQ_COW;
+            break;
+    }
+
+    return cowInfo;
+}
+
+void EnCow_MoveCowsForRandomizer(EnCow* this, GlobalContext* globalCtx) {
+    struct CowInfo cowInfo = EnCow_GetCowInfo(this, globalCtx);
+
+    // Only move the cow body (the tail will be moved with the body)
+    if (this->actor.params != 0) {
+        return;
+    }
+
+    // Move left cow in lon lon tower
+    if (globalCtx->sceneNum == SCENE_SOUKO && this->actor.world.pos.x == -108 && this->actor.world.pos.z == -65) {
+        this->actor.world.pos.x = -229.0f;
+        this->actor.world.pos.z = 157.0f;
+        this->actor.shape.rot.y = 15783.0f;
+    // Move right cow in lon lon stable
+    } else if (globalCtx->sceneNum == SCENE_MALON_STABLE && this->actor.world.pos.x == -3 && this->actor.world.pos.z == -254) {
+        this->actor.world.pos.x += 119.0f;
+    }
+}
+
+void EnCow_SetCowMilked(EnCow* this, GlobalContext* globalCtx) {
+    struct CowInfo cowInfo = EnCow_GetCowInfo(this, globalCtx);
+    gSaveContext.cowsMilked[cowInfo.cowId] = 1;
+}
+
+GetItemID EnCow_GetRandomizerItemFromCow(EnCow* this, GlobalContext* globalCtx) {
```

Comment:
not sure about the naming here

this handles the logic for
* getting the randomized item when we haven't gotten it yet
* getting milk if we have a bottle
* getting ITEM_NONE if neither of those are true

i suggested `EnCow_GetRandomizedItem` earlier, but that doesn't feel quite right. Maybe just `EnCow_GetRandomizerItem` is better but i don't really like that either.

maybe just `EnCow_GetItem` works